kratos/includes/serializer.h

```diff
@@ -225,8 +231,13 @@ class KRATOS_API(KRATOS_CORE) Serializer
             {
                 if(pointer_type == SP_BASE_CLASS_POINTER)
                 {
-                    if(!pValue) {
-                        pValue = Kratos::shared_ptr<TDataType>(new TDataType);
+                    if constexpr (!std::is_abstract_v<TDataType>) {
+                        if(!pValue) {
+                            pValue = Kratos::shared_ptr<TDataType>(new TDataType);
```

Comment:
Avoid double allocations if you know it's an `std::shared_ptr`.

```suggestion
                            pValue = std::make_shared<TDataType>();
```

Reply:
Hi @matekelemen,

Thanks a lot for your review. I completely agree with your suggestion. In fact, I would like to get rid of a few more "naked" `new`s, by using `Kratos::make_intrusive` and `std::make_unique`. However, when I tried your suggestion, compilation failed since a few classes don't have a publicly accessible default constructor (see classes `NurbsCurveGeometry`, `NurbsSurfaceGeometry`, and `NurbsVolumeGeometry`). As a result, `std::make_shared` cannot construct instances of these classes. I guess that the @KratosMultiphysics/technical-committee needs to decide about the desired behavior. If they would like to keep the constructors of the aforementioned classes `private`, using `std::make_shared` is not an option. If, however, they feel it is okay to make these constructors `public`, we can change the implementation as per your suggestion.

Reply:
Oh that's nasty. In that case, you can do smth like [this](https://godbolt.org/#z:OYLghAFBqd5TKALEBjA9gEwKYFFMCWALugE4A0BIEAZgQDbYB2AhgLbYgDkAjF%2BTXRMiAZVQtGIHgBYBQogFUAztgAKAD24AGfgCsp5eiyahUAUgBMAIUtXyKxqiIEh1ZpgDC6egFc2TEAA2cncAGQImbAA5PwAjbFIQC1kAB3QlYhcmL19/IPI0jOchcMiYtnjE5PtsR2KmESIWUiIcvwDgh2wnLMbmolLouISk2SUmlra8zomBiKGKkeSASnt0H1JUTi5LAGYI1F8cAGozXY8ONjIATzPcMy0AQT2Do%2BxT85dx0mx2O4fnk9vj4nMdHqcAOw2CEAETONieAMOLCUSmOCOeUIBxxxxxopAI7mOwNBIgSBAkBAAXgl4djcVYIMtIdC4YjYXTEUCiKQQURjmSCZSaaQAWYsU9ccciNg2CkjDKPh5kajjgAVf6S3HjTAgEBKJDNbCYAD6KR5Zw8Gt2uGOAFkWABrNQ8pn0nHijFSqUEGjHDBMcbYdQpUjHCA6vUEJQmnA0Fg%2BehEE0BknOWKME0AN0t1twzM97u9UsjaHW/MtltOFgsbCd2BNBqNmDMAFYPExLBZOeDi96fkQNkxiURdSA687G4afi3znm3bsvX3xTDjrUVCyi32R2OMD4K%2Bcq13IgB3Nsdrs97dSgdDnd6pszs0Wud3CCn9XLK/Ljlaj2/zE4UXMUngifk6wiJlNz/AVyWFBJiR7ItSz3A8PCPGtwXjJwbhAatu2AmCExIY4UhNFgPlXJQADoHWdVQXw8Z4bQXDFkNHPVUKVDDrDxFgcNIa48MvQjexxYj0FIk1Yko4laPrBjRXOGwWK/USVwBLhVnobhW34AIuB0ch0G4dDrFsYl1k2d49j4cgiG0LTVkdEBWy0QxuGkfg2Fc9yDKMkyuH4JQQHchzDK08g4FgFBg26fcyEoahmmAJRVGMWoRCQdATwMuyMDlBgWHqdLInoLKcv8/gCpSBgRmAGQLHIGq6tIKJ2G2ar0EKxhSAAeX3CrcsckJ1G6R5SFS7h%2BDi1BGkIAz%2BEEYQxAkTgZDkYRlDUTQIvIfRdkMYxTFsWxDAIWIQsgVZ0HNLIQq4ABaHUzhhcxzOsHgIX4dAswSAkcCuplyF5YQCA4NV0G8IHViUKytgMHUIlKzLsuG3h%2BBPUgWBSbg%2BG03T9JGwLcDG1AErDdQAA5AkewJpGOYBUFQY4ZGoixwzMmxrHIY58GIMhq12Hhln4cKdGWVYkF%2BHBEmBly3I8rgvPIKrjOm%2BxQvsxzJaVixvN81XiY18XIuihB4AgWKyYppKIBStKMvKtG8q6nriqyFHncqkaWt6tAjBME0eR8JhHWa7rat69qOA1v2EgG0QXZG2aJqmoLRu6eaIg15bRHESQNrz7aNBG/QmsDk6PrsegLph4y7qEB7ntHV73u5qwvp%2Bv7SAB7B69B5wIah%2Bh67hjYEZ4exR2Rp2htd8gsZxvHIp0rg9KNvaSZtkhKZpumGeRExwxDsPmQgLmzr5whd6FkWxZ15zDbXlWfMVtXAuCrXTdWc2rZAWatsqD20mo7Mq88V4R3diVOeyc9rx0SFmVAKQyJZh4AAThNAqfuyZqbBAQTHTqUCo4J0GnAoyqdQEa1mtnRam185rSkLIYuKhS57X0FPSuaBTo81rpdeAN1G6Bm4AAekjO3WwXdjI9z7gPUOQ9sCQ2hgItYE91rTwWl7CBGNF7Y1xhjAm68iZb24KTeKt88HHDYEoJBxw0HoOotg8YnMeF2GvgLMMtlRbawirraWLBZbUCforF%2BxiAoaxCmFR%2BesDbv2NhnH%2BhjdhhJ%2BibaJPcMiuGkEAA%3D) in the meantime.